great_expectations/datasource/data_connector/configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,101 @@
+import logging
+import os
+from typing import List, Optional
+
+try:
+    from azure.storage.blob import BlobServiceClient
+except ImportError:
+    azure.storage.blob = None
+
+from great_expectations.core.batch import BatchDefinition
+from great_expectations.core.batch_spec import AzureBatchSpec, PathBatchSpec
+from great_expectations.datasource.data_connector import (
+    ConfiguredAssetFilePathDataConnector,
+)
+from great_expectations.datasource.data_connector.asset import Asset
+from great_expectations.datasource.data_connector.util import list_azure_keys
+from great_expectations.execution_engine import ExecutionEngine
+
+logger = logging.getLogger(__name__)
+
+
+class ConfiguredAssetAzureDataConnector(ConfiguredAssetFilePathDataConnector):
+    def __init__(
+        self,
+        name: str,
+        datasource_name: str,
+        container: str,
+        assets: dict,
+        execution_engine: Optional[ExecutionEngine] = None,
+        default_regex: Optional[dict] = None,
+        sorters: Optional[list] = None,
+        name_starts_with: str = "",
+        delimiter: str = "/",
+        max_keys: int = 1000,  # TODO(cdkini): Do we need to address this in Azure?
+        azure_options: Optional[dict] = None,
+        batch_spec_passthrough: Optional[dict] = None,
+    ):
+        logger.debug(f'Constructing ConfiguredAssetAzureDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            assets=assets,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+        self._container = container
+        self._name_starts_with = os.path.join(name_starts_with, "")
+        self._delimiter = delimiter
+        self._max_keys = max_keys
+
+        if azure_options is None:
+            azure_options = {}
+
+        try:
+            if "conn_str" in azure_options:
+                self._azure = BlobServiceClient.from_connection_string(**azure_options)
+            else:
+                self._azure = BlobServiceClient(**azure_options)
```

Comment:
If `azure_options` is `None`, we will always error out due to lacking proper credentials? I'm not sure this check is doing too much.